test/unit/api_spec.js

```diff
@@ -878,6 +878,20 @@ describe("api", function () {
 
       await loadingTask.destroy();
     });
+
+    it("Don't iterate over all empty slots in the xref entries (bug 1980958)", async function () {
+      if (isNodeJS) {
+        pending("Worker is not supported in Node.js.");
+      }
+      const loadingTask = getDocument(buildGetDocumentParams("bug1980958.pdf"));
+      const { promise, resolve } = Promise.withResolvers();
+      setTimeout(() => resolve(null), 1000);
+
+      const pdfDocument = await Promise.race([loadingTask.promise, promise]);
+      expect(pdfDocument?.numPages).toEqual(1);
+
+      loadingTask._worker.destroy();
```

Comment:
Can we use `await loadingTask.destroy();` here like in the other tests? That is public API which prevents having to reach into the private worker property.